src/protocol/attribution/mod.rs

```diff
@@ -69,6 +69,7 @@ pub async fn prefix_or_binary_tree_style<F, C, S>(
     ctx: C,
     stop_bits: &[S],
     uncapped_credits: &[S],
+    should_add_on_first_iteration: bool,
```

Comment:
OK, I think I understand this now, but it's poorly documented (my fault, not yours). I think we should address this with a comment.

```
/// If the caller has foreknowledge that there will never be any two adjacent rows, *both* containing a 1, then it is same to pass `true`, which will simply add values on the first iteration (thereby saving one multiplication per row). If the caller does not know of any such guarantee, `false` should be passed.